### PR TITLE
Code Improvements (Part 1)

### PR DESCRIPTION
**Pull request by Tommy-XL**
Don't merge this until I'm done, mostly I need to see the changes I've made to the code and also someone can suggest ideas for improvement

### CODE REVIEW

File: Patches/CheckGameEndPatch.cs [hunk 1]
```diff
@@ -491,7 +491,7 @@ public bool CheckGameEndByLivingPlayers(out GameOverReason reason)
         {
             reason = GameOverReason.ImpostorByKill;
 
-            if (CustomRolesHelper.RoleExist(CustomRoles.Sunnyboy) && Main.AllAlivePlayerControls.Count() > 1) return false;
+            if (CustomRolesHelper.RoleExist(CustomRoles.Sunnyboy) && Main.AllAlivePlayerControls.Any()) return false;
```
Comment (Gurge44) [inline Patches/CheckGameEndPatch.cs:494]: This needs to be reverted cause count > 1 isn't the same as Any.
> 1 means 2 or more, while Any means 1 or more

dw I ran into the same problem 🤣

Comment (0xDrMoe) [reply Patches/CheckGameEndPatch.cs:494]: Could we not just do >= 1?

Comment (Gurge44) [reply Patches/CheckGameEndPatch.cs:494]: No cause it has to check if there's anyone else other than the sunnyboy alive, aka the player count is higher than 1 or not

Comment (Tommy-XL) [reply Patches/CheckGameEndPatch.cs:494]: Yes, I'll fix it, I didn't pay attention to it at the time

Comment (Tommy-XL) [reply Patches/CheckGameEndPatch.cs]: Fixed

**Review by Gurge44 — APPROVED** [review]

**Comment by 0xDrMoe** [comment]
ETA on this @Tommy-XL ?

**Comment by Tommy-XL** [comment]
Well, this PR can be leaked right now, if necessary

**Comment by Tommy-XL** [comment]
I won't have much time today or tomorrow to finish this, but I'll call it part 1. If you need to merge this as soon as possible then you can do it right now.  And when I have time, I'll just open a new PR and call it part 2 (if the current PR will be merged)

**Pull request merged**